packages/wallets/src/components/FormDropdown/FormDropdown.tsx

```diff
@@ -0,0 +1,64 @@
+import React, { ComponentProps, forwardRef, Ref } from 'react';
+import { Field, FieldProps } from 'formik';
+import * as Yup from 'yup';
+import { Dropdown } from '@deriv-com/ui';
+import './FormDropdown.scss';
+
+type DropdownProps = Omit<ComponentProps<typeof Dropdown>, 'onSelect'>;
+
+export type TFormDropdownProps = DropdownProps & {
+    name: string;
+    onSelect?: (selectedValue: string) => void;
+    validationSchema?: Yup.AnySchema;
+} & { ref: Ref<HTMLInputElement> };
+
+const FormDropdown = forwardRef(
+    ({ disabled, isRequired, name, onSelect, validationSchema, ...rest }: TFormDropdownProps) => {
+        const validateField = (value: unknown) => {
+            try {
+                if (validationSchema) {
+                    validationSchema.validateSync(value);
+                }
+            } catch (err: unknown) {
+                return (err as Yup.ValidationError).message;
+            }
+        };
+        return (
+            <Field name={name} validate={validateField}>
+                {({ field, form }: FieldProps) => {
+                    const isFieldInvalid = Boolean(form.touched[name] && isRequired && form.errors[name]);
+                    return (
+                        <div className='wallets-form-dropdown'>
+                            <Dropdown
+                                {...rest}
+                                disabled={disabled}
+                                errorMessage={isFieldInvalid ? (form.errors[name] as string) : ''}
+                                name={name}
+                                onBlur={() => {
+                                    if (!form.touched[name]) {
+                                        form.setFieldTouched(name);
+                                    }
+                                }}
```

Comment:
@aum-deriv 
do we really need to set touched if we do the same thing in onSelect ? 🤔
in my understanding there cannot be onBlur without selection first 🤔

Reply:
Hi @heorhi-deriv,
This is done to handle the case when the user just focuses and then unfocuses the dropdown. In this case we need to trigger an error as the field was touched and no value was chosen.